src/Provider.tsx

```diff
@@ -0,0 +1,42 @@
+import React, {
+  useState,
+  createContext,
+  Dispatch,
+  SetStateAction,
+} from 'react';
+import { CacheTimeouts } from './types';
+
+type Props = {
+  children: JSX.Element;
+  timeTillCacheInvalidate?: number;
+};
+
+type ContextProps = {
+  cacheTimeouts: CacheTimeouts;
+  setCacheTimeouts: Dispatch<SetStateAction<CacheTimeouts>>;
+  timeTillCacheInvalidateGlobal?: number;
+};
+
+const ReactUseFetchWithReduxContext = createContext({
+  cacheTimeouts: {},
+  setCacheTimeouts: () => {},
+} as ContextProps);
```

Comment:
I just looked at the typings for `createContext` and I think you can do this:

```
const ReactUseFetchWithReduxContext = createContext<ContextProps>({
  cacheTimeouts: {},
  setCacheTimeouts: () => {},
});
```

Which saves us doing the cast